march_hardware/include/march_hardware/MarchRobot.h

```diff
@@ -0,0 +1,70 @@
+// Copyright 2019 Project March.
+
+#ifndef MARCH4CPP__MARCH4_H
+#define MARCH4CPP__MARCH4_H
+
+#include <string>
+#include <vector>
+#include <stdint.h>
+
+#include <march_hardware/Joint.h>
+
+#include <march_hardware/EtherCAT/EthercatMaster.h>
+
+namespace march4cpp
+{
+class MarchRobot
+{
+private:
+  std::unique_ptr<EthercatMaster> ethercatMaster;
+
+public:
+  ::std::vector<Joint> jointList;
+
+  MarchRobot(::std::vector<Joint> jointList, ::std::string ifName, int ecatCycleTime);
+
+  MarchRobot();
+
+  void startEtherCAT();
+
+  void stopEtherCAT();
+
+  int getMaxSlaveIndex();
+
+  bool hasValidSlaves();
+
+  bool isEthercatOperational();
+
+  Joint getJoint(::std::string jointName);
+
+  /** @brief Override comparison operator */
+  friend bool operator==(const MarchRobot& lhs, const MarchRobot& rhs)
+  {
+    if (lhs.jointList.size() != rhs.jointList.size())
+    {
+      return false;
+    }
+    for (unsigned int i = 0; i < lhs.jointList.size(); i++)
+    {
+      const march4cpp::Joint lhsJoint = lhs.jointList.at(i);
+      const march4cpp::Joint rhsJoint = rhs.jointList.at(i);
+      if (lhsJoint != rhsJoint)
+      {
+        return false;
+      }
+    }
+    return true;
```

Comment:
The tests will not break if false is true

Reply:
Ah correct, I have forgotten to implement a test where we compare unequal objects. That is the case for all objects in march_hardware. While I agree it should be implemented, it is not the responsibility of this pull request. I'll create an issue for it.